nspl/op.php

```diff
@@ -168,3 +168,18 @@ function methodCaller($method, array $args = array())
         return call_user_func_array(array($object, $method), $args);
     };
 }
+
+/**
+ * Returns a function that returns a new instance of a predefined class, passing its parameters to the constructor
+ * @param string $class Class name
+ * @return callable
+ */
+function instanceCreator($class)
+{
+    args\expects(args\string, $class);
+    $reflection_class = new \ReflectionClass($class);
+
+    return function() use ($class, $reflection_class) {
+        return call_user_func_array(array($reflection_class, 'newInstance'), func_get_args());
+    };
+}
```

Comment:
NSPL is very focused on speed. There are two optimizations which could be made:
1. Starting PHP 5.6 we have the splat operator ```...```. Using we can get rid of reflection which is relatively slow and have the following code ```return new $class(...func_get_args());```. In my own tests, it makes the speed about two times faster.
2. Again the splat operator is faster than a function call so we can get rid of ```func_get_args()``` too. According to my tests, it makes the code about 2.5 times faster than the original version.

So the final code can be something like that:
```php
function instanceCreator($class)
{
    args\expects(args\string, $class);

    return function() use ($class) {
        return new $class(...func_get_args());
    };
}
```

We can use ```PHP_VERSION``` or ```PHP_MAJOR_VERSION``` and ```PHP_MINOR_VERSION``` constants to provide a PHP 5.3 solution too.